### ISSUE TITLE
Commit messages of the rewrite

### ISSUE THREAD

**Issue by stappersg** [pusherofbrooms/avr-delay#22]
In #17 are many things discussed. None of that information is in branch `rewrite`. That shouldn't be.

Have look at https://github.com/ipxe/ipxe/commits for examples of good commit messages.

Thing I'm asking for is commit messages, not only commit summaries.

Rule of thumb:  More than one line.

Please provide the missing messages for branch `rewrite`.

**Comment by lord-ne** [comment]
How do I add the messages to commits that already exist?

**Comment by stappersg** [comment]
On Thu, Jun 09, 2022 at 03:53:00PM -0700, lord-ne wrote:
> How do I add the messages to commits that already exist?
 
I can do that.  (It will be done outside "github.com".)


It is the text that this issue should yield.  Paste them here.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by lord-ne** [comment]
Here is the commit message for the first commit ([Add delay_count and delay_loop internal functions](https://github.com/avr-rust/delay/commit/b5358bab1396bf3024f889dcb40a35ada5aa8412)):

```
Move delay loop into internal functions

Refactor the assembly loop into its own function, to allow easier
reuse in the below functions. 

Add internal delay functions calling the assembly loop function.
One function uses a 32 bit counter, and the other uses a 48 bit counter.
These functions will be used to reactor delay_ms and delay_us to prevent
overflow. 

Refactor delay to use the internal delay functions.
```

Here is the commit message for the second commit ([Implement delay, delay_ms, and delay_us](https://github.com/avr-rust/delay/commit/0909757a8b98d743fff583bd93752bf48f3508a4)):

```
Refactor delay_ms and delay_us to prevent overflow

Change delay_ms and delay_us to call the internal delay function with
a 48 bit counter. This prevents delay_ms and delay_us from overflowing
when passed a large argument.

Move the internal delay functions to their own module. Remove the
unused internal delay_loop_3_cycles function.
```

Here is the commit message for the third commit ([Divide numerator and denominator by their GCD at compile time](https://github.com/avr-rust/delay/commit/64ff9ba4dd6750588da8e152fcd0f56f837986f4)):

```
Divide numerator and denominator by their GCD at compile time

Calculating the number of cycles to delay for in delay_ms and delay_us
requires multiplying by a constant and dividing by another constant. Divide
both constants by their GCD at compile time, to allow the compiler to
better optimize the multiplication and division. This reduces the overhead in
the case where the compiler does not know the length of the delay at
compile time, or otherwise fails to inline the function.
```

**Comment by stappersg** [comment]
Those commit message are now at http://uta.stappers.it/git/avr-rust/delay in branch `rerewrite` use something like
```text
git remote add stappers http://uta.stappers.it/git/avr-rust/delay
git fetch stappers
git checkout rerewrite 
```
to get them.  File `src/lib.rs` now still contains
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        assert_eq!(2 + 2, 4);
    }
}
```
For those who like to remove that `#[cfg(test)]`, ask for it a separate merge request. Thing I'm really telling is that I don't like silent removal of code.  Addtional note:  _no hard feelings_.  `:-)`

@lord-ne if you want the _"rewrite changes"_, that originate from #17, merged, make a _merge request_ so we can discuss this further. Branch `rerewrite` is created for that purpose.

**Comment by lord-ne** [comment]
I created the pull request, so I think this issue can be closed now

**Comment by stappersg** [comment]
merged  (now closing the issue)
